### PR TITLE
MeshAttributes are no longer bit-wise flags

### LINKED ISSUE #286: Remove bitwise ops for MeshAttribute enum
The MeshAttribute enum uses bitwise operations to provide support for combinations of mesh types. Unfortunately, this is overly complicated and buggy. We need to convert this to a standard enum and explicitly handle the (limited) mesh combinations.

### PR DESCRIPTION
**Pull request by andrsd**
- MeshAttributes are not bit-wise flags
- Removing trailing underscores from public member variables in SplitMeshInfo
- Creating setters according to code standards in `UnpartitionedMesh`

Closes #286

### CODE REVIEW

**Review by zhardy-lanl — COMMENTED** [review]
Some more work needs to be done here to finalize this. First, `MeshAttributes` should probably be renamed to something like `MeshType`. Attributes was used as a catchall term. Second, a mesh can be unstructured or orthogonal **and** extruded, so this structure doesn't quite work I don't think.

Here is my proposal:
```
enum MeshType : int
{
  UNSTRUCTURED,
  ORTHOGONAL
}
```
Meshes should be defaulted to `UNSTRUCTURED` since that is the most general.

Additionally, a bool attribute should be added to `MeshContinuum` called `extruded_`, defaulted to false, with a getters and setters for now.

**Review by wdhawkins — APPROVED** [review]

**Comment by andrsd** [comment]
I can do this work in this PR. It is just that #286 did not specify this work - just undo the bit-wise flags. But, I do agree with your comment @zhardy-lanl .

**Comment by ragusa** [comment]
> I can do this work in this PR. It is just that #286 did not specify this work - just undo the bit-wise flags. But, I do agree with your comment @zhardy-lanl .

Sorry the specs were not clear in the issue. I agree with Zach and thanks for updating this PR.

File: framework/mesh/mesh_generator/split_file_mesh_generator.cc [hunk 1]
```diff
@@ -316,12 +317,13 @@ SplitFileMeshGenerator::ReadSplitMesh()
                          std::to_string(opensn::mpi_comm.size()) + " processes.");
 
   info_block.dimension = ReadBinaryValue<unsigned int>(ifile);
-  info_block.mesh_attributes_ = ReadBinaryValue<int>(ifile);
+  info_block.mesh_type = static_cast<MeshType>(ReadBinaryValue<int>(ifile));
```
Comment (ragusa) [inline framework/mesh/mesh_generator/split_file_mesh_generator.cc:320]: just curious what the static cast does here. it wasn't there before. what is an mistake?

**Review by ragusa — APPROVED** [review]

File: framework/mesh/mesh_generator/split_file_mesh_generator.cc (hunk 1 above)
Comment (andrsd) [reply framework/mesh/mesh_generator/split_file_mesh_generator.cc:320]: Originally `mesh_attributes_` were declared as `int` the `SplitMeshInfo` and the static cast was done `SplitFileMeshGenerator::SetupLocalMesh `. I changed the declaration of `SplitMeshInfo::mesh_attributes_` to be `MeshAttributes`, so the static cast was moved inside the `SplitFileMeshGenerator::ReadSplitMesh `.

Then, `mesh_attributes_` became `mesh_type`...

File: framework/mesh/unpartitioned_mesh/unpartitioned_mesh.h [hunk 2]
```diff
@@ -64,11 +64,14 @@ class UnpartitionedMesh
...
-  Options& MeshOptions() { return mesh_options_; }
+  void SetMeshOptions(const Options& opts) { mesh_options_ = opts; }
   const Options& MeshOptions() const { return mesh_options_; }
 
-  MeshAttributes& Attributes() { return attributes_; }
-  const MeshAttributes& Attributes() const { return attributes_; }
+  void SetType(MeshType type) { mesh_type_ = type; }
+  const MeshType& Type() const { return mesh_type_; }
+
+  void SetExtruded(bool extruded = true) { extruded_ = extruded; }
```
Comment (zhardy-lanl) [inline framework/mesh/unpartitioned_mesh/unpartitioned_mesh.h:73]: While it makes sense to have this default to `true`, this routine appearing without an argument looks a little funny. Would it make more sense to just not have a default argument?

**Review by zhardy-lanl — APPROVED** [review]

**Pull request merged**